### PR TITLE
Lua_api.txt: Add documentation of required mapgen aliases

### PR DESCRIPTION
**Pull request by paramat**
Attends to #5988 and part of #5854

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -235,6 +236,56 @@ you have an item called `epiclylongmodname:stuff`, you could do
 
 and be able to use `/giveme stuff`.
 
+Mapgen aliases
```
Comment (rubenwardy) [inline doc/lua_api.txt]: `###` to be a subheading

Comment (paramat) [reply doc/lua_api.txt]: I think it's important enough to have it's own section, makes it visually easier to find too.

**Comment by paramat** [comment]
Updated and improved.

**Comment by paramat** [comment]
4d83eb796b317901df3b2be8f3c8ace339f9bb95

**Pull request closed without merging**